From bebfa0ffc10c8b6fc6d893114d1969625cb019e2 Mon Sep 17 00:00:00 2001
From: Naga Irrinki <naga.irrinki@broadcom.com>
Date: Tue, 9 Feb 2016 14:30:26 +0530
Subject: [PATCH] be2net: backport for RHEL 7.2

Adding the changes required for RH 7.2. Using
HAVE_NDO_DFLT_BRIDGE_GETLINK_7_PARAMS macro.

Signed-off-by: Naga Irrinki <naga.irrinki@broadcom.com>
Signed-off-by: Selvin Xavier <selvin.xavier@broadcom.com>
---
 drivers/net/ethernet/emulex/benet/be_main.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index f5d6672..d0a8364 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -4606,7 +4606,11 @@ static int be_ndo_bridge_getlink(struct sk_buff *skb, u32 pid, u32 seq,
 
 	return ndo_dflt_bridge_getlink(skb, pid, seq, dev,
 				       hsw_mode == PORT_FWD_TYPE_VEPA ?
-				       BRIDGE_MODE_VEPA : BRIDGE_MODE_VEB);
+				       BRIDGE_MODE_VEPA : BRIDGE_MODE_VEB
+#ifdef HAVE_NDO_DFLT_BRIDGE_GETLINK_7_PARAMS
+                                        , 0, filter_mask
+#endif
+					);
 }
 
 #endif
-- 
1.8.3.1

